src/_macosx.m

```diff
@@ -934,20 +938,19 @@ -(void)save_figure:(id)sender { gil_call_method(toolbar, "save_figure"); }
     [self->handler installCallbacks: actions forButtons: buttons];
 
     NSFont* font = [NSFont systemFontOfSize: 0.0];
-    rect.size.width = 300;
-    rect.size.height = 0;
-    rect.origin.x += height;
-    NSTextView* messagebox = [[NSTextView alloc] initWithFrame: rect];
+    // rect.origin.x is now at the far right edge of the buttons
+    // we want the messagebox to take up the rest of the toolbar area
+    rect.size.width = bounds.size.width - rect.origin.x;
```

Comment:
Is this fine if the buttons end past the bounds (so this is a negative width)? Is that not allowed?

Reply:
Do you mean if I shrink the figure to less than the width of the buttons on the toolbar?  I did that just now, and while you can't see the cursor output anymore, everything still runs, and widening again still has the cursor output.

Reply:
Yes, also what if the windows _starts_ smaller than the toolbar would normally be?

Reply:
That seems fine as well (figsize=(2, 5))

Reply:
I mean we could add a trap here, but I don't see that anything is breaking by passing in negative width.

Reply:
negative width's apparently are allowed, but have potentially undefined behavior later on. I just put a max(calc, 0) in front of it to make a 0 minimum.

While playing around with this, I also noticed that resizing the window reset the disableCursorRects piece and caused the same issues as before. To fix that I moved the disableCursorRects into the messagebox code piece as well since that is the piece we want to disable it from.